File: usaco/bronze/outofplace.py
import sys
# sys.stdin = open('input.txt', 'r')
sys.stdin = open('outofplace.in', 'r')
sys.stdout = open('outofplace.out', 'w')

n = int(input())
line = [int(input()) for x in range(n)]

sorted_line = sorted(line)
ans = -1
for i in range(n):
    if sorted_line[i] != line[i]:
        ans += 1
ans = max(0, ans)
print(ans)

File: usaco/bronze/outofplace.cpp
#include <bits/stdc++.h>
#define ll long long
using namespace std;

void setIO(string s) {
  ios_base::sync_with_stdio(0); cin.tie(0); 
  freopen((s + ".in").c_str(), "r", stdin);
  freopen((s + ".out").c_str(), "w", stdout);
}

int main() {
  setIO("outofplace");
  ll n, ans = 0; cin >> n;
  vector<ll> A(n), B(n);
  for (ll i = 0; i < n; ++i) { cin >> A[i]; B[i] = A[i]; }
  sort(B.begin(), B.end());
  for (ll i = 0; i < n; ++i) {
    if (A[i] != B[i]) ans++;
  }
  cout << ans - 1;
  return 0;
}